chart/templates/deployment.yaml

```diff
@@ -0,0 +1,100 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "s3proxy.fullname" . }}
+  labels:
+    {{- include "s3proxy.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 2        # how many pods we can add at a time
+      maxUnavailable: 0   
+  selector:
+    matchLabels:
+      {{- include "s3proxy.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "s3proxy.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "s3proxy.serviceAccountName" . }}
+      securityContext:
+        {{ .Values.podSecurityContext.runAsUser  }}
+        {{ .Values.podSecurityContext.runAsGroup  }}
+        {{ .Values.podSecurityContext.fsGroup  }}
+        {{- if semverCompare ">=1.23-0" .Capabilities.KubeVersion.GitVersion -}}
+        {{ .Values.podSecurityContext.fsGroupChangePolicy  }}
+        {{- end }}
+      containers:       
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 8000
+              protocol: TCP
+          # Note in general use these would fail tracking request for health probe https://github.com/gaul/s3proxy/issues/427
+          # livenessProbe:
```

Comment:
Consider using a TCP liveness probe in the meantime.

Reply:
TCP liveness probes open connections and don't send data. I have seen apps that have issues with that I didn't want to add it in this PR without testing.